pkg/kubectl/cmd/get/get.go

```diff
@@ -345,6 +345,9 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
 	if err != nil {
 		allErrs = append(allErrs, err)
 	}
+	if multipleGVKsRequested(infos) {
+		o.PrintFlags.EnsureWithKind()
```

Comment:
don't mutate o.PrintFlags here, set a local withKind boolean, pass it into ToPrinter, and set it on the copy of the print flags inside that function